dtonator/src/main/java/com/bizo/dtonator/GenerateEnum.java

```diff
@@ -44,6 +52,29 @@ private void addEnumValues() {
     }
   }
 
+  private void addEnumGetters() {
+    try {
+      final Class<?> clazz = Class.forName(dto.getDomainType());
+      for (final Field field : clazz.getDeclaredFields()) {
+        if (!field.isEnumConstant() && !field.isSynthetic()) {
+          final GMethod getter = gc.getMethod("get{}", capitalize(field.getName())).returnType(field.getType());
+          getter.body.line("switch (this) {");
+          for (final String value : dto.getEnumValues()) {
+            @SuppressWarnings({ "unchecked", "rawtypes" })
+            final Enum enumClass = Enum.valueOf((Class<? extends Enum>) clazz, value);
```

Comment:
Maybe just "enum" or "enumInstance" instead of "enumClass"? Because if I'm reading it right, it's sounds like it's the specific instance/value of an enum type.